src/amount.cpp

```diff
@@ -19,10 +19,10 @@ CFeeRate::CFeeRate(const CAmount& nFeePaid, size_t nSize)
 
 CAmount CFeeRate::GetFee(size_t nSize) const
 {
-    CAmount nFee = nSatoshisPerK*nSize / 1000;
+    CAmount nFee = nSatoshisPerK * nSize / 1000;
 
-    if (nFee == 0 && nSatoshisPerK > 0)
-        nFee = nSatoshisPerK;
+    if (nFee == 0 && nSize != 0 && nSatoshisPerK != 0)
```

Comment:
@morcos Actually, I don't think we support negative amounts for `nSatoshisPerK` right now. 64-bit platforms will convert `nSatoshisPerK` to `unsiged` (`size_t`). 32-bit platforms will convert `nSize` to signed `CAmount`.